monkey/monkey_island/cc/island_event_handlers/handle_agent_registration.py

```diff
@@ -23,10 +24,12 @@ def __init__(
         machine_repository: IMachineRepository,
         agent_repository: IAgentRepository,
         node_repository: INodeRepository,
+        get_current_datetime: Callable[[], datetime] = datetime.now(),
```

Comment:
Why callable ?

Reply:
This is in order to get around the inability to monkeypatch `datetime.now`

Reply:
In which tests do we need to monkeypatch it? Because I really don't like if we do it just for the tests.

Reply:
It was updated in order to test that the agent registration handler initializes the Agent appropriately

Reply:
It's not ideal, but it's the best solution we came up with. It gives us some degree of flexibility. For example, we could chose to construct this component with either `datetime.now()` or `datetime.utcnow()`, or any other equivalent function. I don't think we'll use this capability, but we have it :)

Since it's defaulted to `datetime.now`, everything other than tests can just ignore it.